eslint.config.mjs

```diff
@@ -0,0 +1,63 @@
+import { includeIgnoreFile } from '@eslint/compat';
+import { defineConfig, globalIgnores } from 'eslint/config';
+import * as path from 'node:path';
+import { createBaseConfig, createTestConfig } from '@mui/internal-code-infra/eslint';
+import { fileURLToPath } from 'url';
+
+const filename = fileURLToPath(import.meta.url);
+const dirname = path.dirname(filename);
+
+export default defineConfig(
+  includeIgnoreFile(path.join(dirname, '.gitignore')),
+  includeIgnoreFile(path.join(dirname, '.eslintignore')),
+  globalIgnores(
+    ['packages/mui-internal-code-infra/src/eslint/material-ui/rules/*.test.*'],
+    'Global ignores',
+  ),
+  {
+    name: 'Base config',
+    extends: createBaseConfig(),
+    rules: {
+      'import/prefer-default-export': 'off',
+      // No time for this
+      'react/prop-types': 'off',
+      'jsx-a11y/control-has-associated-label': 'off',
+      'jsx-a11y/no-autofocus': 'off',
+    },
+  },
+  {
+    files: [
+      // matching the pattern of the test runner
+      '**/*.test.?(c|m)[jt]s?(x)',
+    ],
+    extends: createTestConfig(),
+  },
+  {
+    files: ['**/*.mjs'],
+    rules: {
+      'import/extensions': ['error', 'ignorePackages'],
+    },
+  },
+  {
+    files: ['**/apps/**/*'],
+    rules: {
+      'import/no-relative-packages': 'off',
+      'react/jsx-one-expression-per-line': 'off',
+    },
+  },
+  {
+    files: ['packages/bundle-size-checker/**/*'],
+    rules: {
+      'import/prefer-default-export': 'off',
```

Comment:
We may consider removing this rule altogether and add a specific override for core to enable it, I believe this would be overridden on base ui as well, right?

Reply:
Ah, there is also https://github.com/mui/mui-public/pull/344/files#diff-9601a8f6c734c2001be34a2361f76946d19a39a709b5e8c624a2a5a0aade05f2R38.

Wouldn't these do the same thing?

Reply:
🤔 oh, wait, I see, these are .js files because type: module